autogpt_platform/backend/backend/blocks/postgres.py

```diff
@@ -0,0 +1,734 @@
+from enum import Enum
+from typing import Any, List, Literal, Optional
+
+import psycopg2
+from psycopg2.extras import RealDictCursor
+from pydantic import BaseModel, SecretStr
+
+from backend.data.block import Block, BlockCategory, BlockOutput, BlockSchema
+from backend.data.model import (
+    CredentialsField,
+    CredentialsMetaInput,
+    SchemaField,
+    UserPasswordCredentials,
+)
+from backend.integrations.providers import ProviderName
+
+PostgresCredentials = UserPasswordCredentials
+PostgresCredentialsInput = CredentialsMetaInput[
+    Literal[ProviderName.POSTGRES],
+    Literal["user_password"],
+]
+
+
+def PostgresCredentialsField() -> PostgresCredentialsInput:
+    """Creates a Postgres credentials input on a block."""
+    return CredentialsField(
+        description="The Postgres integration requires a username and password.",
+    )
+
+
+TEST_POSTGRES_CREDENTIALS = UserPasswordCredentials(
+    id="01234567-89ab-cdef-0123-456789abcdef",
+    provider="postgres",
+    username=SecretStr("mock-postgres-username"),
+    password=SecretStr("mock-postgres-password"),
+    title="Mock Postgres credentials",
+)
+
+TEST_POSTGRES_CREDENTIALS_INPUT = {
+    "provider": TEST_POSTGRES_CREDENTIALS.provider,
+    "id": TEST_POSTGRES_CREDENTIALS.id,
+    "type": TEST_POSTGRES_CREDENTIALS.type,
+    "title": TEST_POSTGRES_CREDENTIALS.title,
+}
+
+
+class CommandType(str, Enum):
+    TRUNCATE = "TRUNCATE"
+    DELETE = "DELETE"
+    DROP = "DROP"
+
+
+class ConditionOperator(str, Enum):
+    EQUALS = "="
+    NOT_EQUALS = "<>"
+    GREATER_THAN = ">"
+    LESS_THAN = "<"
+    GREATER_EQUALS = ">="
+    LESS_EQUALS = "<="
+    LIKE = "LIKE"
+    IN = "IN"
+
+
+class Condition(BaseModel):
+    column: str
+    operator: ConditionOperator
+    value: Any
+
+
+class CombineCondition(str, Enum):
+    AND = "AND"
+    OR = "OR"
+
+
+class PostgresDeleteBlock(Block):
+    class Input(BlockSchema):
+        credentials: PostgresCredentialsInput = PostgresCredentialsField()
+        host: str = SchemaField(description="Database host", advanced=False)
+        port: int = SchemaField(description="Database port", advanced=False)
+        database: str = SchemaField(description="Database name", default="postgres",advanced=False)
+        schema_: str = SchemaField(description="Schema name", default="public",advanced=False)
+        table: str = SchemaField(description="Table name")
+        command: CommandType = SchemaField(
+            description="Command type to execute",
+            default=CommandType.DELETE,
+            advanced=False
+        )
+        conditions: List[Condition] = SchemaField(
+            description="Conditions for DELETE command",
+            default=[],
```

Comment:
```suggestion
            default_factory=list,
```
We were just fixing bugs caused by this 😄  Not using factory makes the default list shared across all instances. I'm not marking other instances of this but there are many.